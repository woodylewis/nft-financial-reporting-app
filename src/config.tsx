const config = {
  INFURA_PROVIDER: 'xxxxxxxxxxxxxxxxxxxxxxxx',
  CONTRACT_ADDRESS: 'xxxxxxxxxxxxxxxxxxxxxxxx',
  WALLET_ADDRESS: 'xxxxxxxxxxxxxxxxxxxxxxxx',
};

export default config;
