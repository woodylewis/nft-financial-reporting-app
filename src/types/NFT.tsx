type NFT = {
  id: string;
  uri: string;
  metaData?: object;
};

export default NFT;
