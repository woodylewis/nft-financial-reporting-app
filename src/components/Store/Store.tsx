import React, { createContext, useContext, useState, FunctionComponent, ReactNode } from 'react';
import PropTypes from 'prop-types';
import NFT from '../../types/NFT';

import Web3 from 'web3';
/* eslint-disable @typescript-eslint/no-var-requires */
const snftABI = require('../../services/abi.json');
import config from '../../config';

const testnet = config.INFURA_PROVIDER;
const web3 = new Web3(new Web3.providers.HttpProvider(testnet));
web3.eth.defaultAccount = config.WALLET_ADDRESS;
const wallet = web3.eth.defaultAccount;
const contract = new web3.eth.Contract(snftABI, config.CONTRACT_ADDRESS);

export const SNNContext = createContext({
  contextContract: contract,
  contextWallet: wallet,
  tokenMap: new Map(),
  tokensLoaded: false,
  /* eslint-disable */
  setToken: (id: string, t: NFT) => {},
  getToken: (id: string): any => {},
  setTokenMetaData: (id: string, metaData: object) => {},
  setTokensLoaded: (state: boolean) => {},
  /* eslint-enable */
});

export const useSNNContext = () => useContext(SNNContext);

type StoreProps = {
  children: ReactNode;
};

export const Store: FunctionComponent<StoreProps> = ({ children }) => {
  const [tokens, setTokens] = useState(new Map());
  const [tokenState, setTokenState] = useState(false);

  const setToken = (id: string, t: NFT) => {
    setTokens(new Map(tokens.set(id, t)));
  };

  // eslint-disable-next-line
  const getToken = (id: string): any => {
    return tokens.get(id);
  };

  const setTokenMetaData = (id: string, metaData: object) => {
    const token = tokens.get(id);
    token['metaData'] = metaData;
  };

  const setTokensLoaded = (tokenState: boolean) => {
    setTokenState(tokenState);
  };

  return (
    <SNNContext.Provider
      value={{
        contextContract: contract,
        contextWallet: wallet,
        tokensLoaded: tokenState,
        tokenMap: tokens,
        setToken: setToken,
        getToken: getToken,
        setTokenMetaData: setTokenMetaData,
        setTokensLoaded: setTokensLoaded,
      }}
    >
      {children}
    </SNNContext.Provider>
  );
};

Store.propTypes = {
  children: PropTypes.element.isRequired,
};
