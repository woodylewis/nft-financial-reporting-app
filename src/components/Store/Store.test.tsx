import * as React from 'react';
import { render, cleanup } from '@testing-library/react';

import { SNNContext } from './Store';

import NFT from '../../types/NFT';
import Web3 from 'web3';
/* eslint-disable @typescript-eslint/no-var-requires */
const snftABI = require('../../services/abi.json');
import config from '../../config';

const testnet = config.INFURA_PROVIDER;
const web3 = new Web3(new Web3.providers.HttpProvider(testnet));
web3.eth.defaultAccount = config.WALLET_ADDRESS;
const wallet = web3.eth.defaultAccount;
const contract = new web3.eth.Contract(snftABI, config.CONTRACT_ADDRESS);

const valueObj = {
  contextContract: contract,
  contextWallet: wallet,
  tokensLoaded: false,
  tokenMap: new Map(),
  setToken: (id: string, t: NFT) => {},
  getToken: (id: string): any => {},
  setTokenMetaData: (id: string, metaData: object) => {},
  setTokensLoaded: (state: boolean) => {},
};

describe('Store component', () => {
  afterEach(cleanup);

  test('renders Provider', () => {
    const { asFragment } = render(<SNNContext.Provider value={valueObj}>Test</SNNContext.Provider>);
    expect(asFragment).toMatchSnapshot();
    expect(SNNContext.Provider).toBeTruthy();
  });
});
