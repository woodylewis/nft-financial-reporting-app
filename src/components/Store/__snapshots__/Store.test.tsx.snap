// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store component renders Provider 1`] = `[Function]`;
